src/vm/eventtrace.cpp

```diff
@@ -6241,6 +6243,79 @@ VOID ETW::LoaderLog::SendModuleEvent(Module *pModule, DWORD dwEventOptions, BOOL
     }
 }
 
+VOID ETW::MethodLog::SendMethodDetailsEvent(MethodDesc *pMethodDesc)
+{
+    CONTRACTL {
+        NOTHROW;
+        GC_NOTRIGGER;
+    } CONTRACTL_END;
+
+    EX_TRY
+    {
+        if(ETW_TRACING_CATEGORY_ENABLED(MICROSOFT_WINDOWS_DOTNETRUNTIME_PROVIDER_DOTNET_Context, 
+                                        TRACE_LEVEL_INFORMATION, 
+                                        CLR_METHODDIAGNOSTIC_KEYWORD))
+        {
+            if (pMethodDesc->IsDynamicMethod())
+                goto done;
+
+            Instantiation inst = pMethodDesc->GetMethodInstantiation();
+
+            if (inst.GetNumArgs() > 1024) // ETW has a limit for maximum event size. Do not log overly large method type argument sets
+                goto done;
+
+            BulkTypeEventLogger typeLogger;
+
+            ULONGLONG typeID = (ULONGLONG)pMethodDesc->GetMethodTable_NoLogging();
+            ETW::TypeSystemLog::LogTypeAndParametersIfNecessary(&typeLogger, typeID, ETW::TypeSystemLog::kTypeLogBehaviorTakeLockAndLogIfFirstTime);
+            ULONGLONG loaderModuleID = (ULONGLONG)pMethodDesc->GetLoaderModule();
+
+            // We're about to recursively call ourselves for the type parameters, so make a
```

Comment:
I think this is a copy pasted comment that describes a problem that was relevant at the source but not relevant here? As best I can tell the only reason to copy the instantiation into an array is because FireEtwMethodDetails expects an array parameter.

Recommend deleting or updating the comment.